hw/ip/otbn/util/check_call_stack.py

```diff
@@ -28,33 +29,34 @@ def _check_call_stack_insn(insn: Insn, operands: Dict[str, int]) -> bool:
     return True
 
 
-def check_call_stack(program: OTBNProgram) -> Tuple[bool, str]:
+def check_call_stack(program: OTBNProgram) -> CheckResult:
     '''Check that the special register x1 is used safely.
 
     If x1 is used for purposes unrelated to the call stack, it can trigger a
     CALL_STACK error. This check errors if x1 is used for any other instruction
     than `jal` or `jalr`.
     '''
+    out = CheckResult()
     imem_start = program.get_pc_at_symbol('_imem_start')
     imem_end = program.get_pc_at_symbol('_imem_end')
     for pc in range(imem_start, imem_end, 4):
         insn = program.get_insn(pc)
         operands = program.get_operands(pc)
         if not _check_call_stack_insn(insn, operands):
-            return (False, 'check_call_stack: FAIL at PC {:#x}: {} {}'.format(
-                pc, insn.mnemonic, operands))
-    return (True, 'check_call_stack: PASS')
+            out.err(
+                'Potentially dangerous use of the call stack register x1 at PC {:#x}: {}'
+                .format(pc, insn.disassemble(pc, operands)))
+    return out
 
 
 def main() -> int:
     parser = argparse.ArgumentParser()
     parser.add_argument('elf', help=('The .elf file to check.'))
     args = parser.parse_args()
     program = decode_elf(args.elf)
-    ok, msg = check_call_stack(program)
-    print(msg)
-    if not ok:
-        return 1
+    result = check_call_stack(program)
+    result.add_prefix('check_call_stack: ')
+    print(result)
```

Comment:
Maybe the prefix should be added in the `check_call_stack` function? It would make this code a bit simpler looking, I think. Also, don't you want to return 1 if you find something that looks wrong?